src/argilla_server/schemas/v1/users.py

```diff
@@ -34,5 +38,13 @@ class Config:
         orm_mode = True
 
 
+class UserCreate(BaseModel):
+    first_name: constr(min_length=1, strip_whitespace=True)
+    last_name: Optional[constr(min_length=1, strip_whitespace=True)]
+    username: str = Field(regex=USER_USERNAME_REGEX, min_length=1)
```

Comment:
`regex` has been deprecated and replaced by `pattern` a long time ago.

```suggestion
    username: str = Field(pattern=USER_USERNAME_REGEX, min_length=1)
```

Reply:
Not sure if it's our Pydantic version but it's not working using `pattern` parameter. It's not working even using `r` sigil to specify the regular expression like:

```python
USER_USERNAME_PATTERN = r"^(?!-|_)[A-za-z0-9-_]+$"
```